### ISSUE TITLE
Incorrect fix for variable being used as a loop counter

### ISSUE THREAD

**Issue by wainwrightmark** [rust-lang/rust-clippy#9013]
### Summary

I think the automatic fix for an index being used as a loop counter is broken. My guess is that it ought to be removing the original variable declaration and the increment statement. I'm getting the same error on both stable and nightly.

### Reproducer

I tried this code:

```rust
fn main() {
    let vector = vec!['a', 'b','c'];
    let mut index:usize = 0;
    
    for v in vector{
        println!("{v} {index}");
        index+=1;
    }
}
```

I ran `cargo clippy --fix` expecting either no change or refactoring to use `.into_iter().enumerate()` 

Instead, this happened:

```
warning: failed to automatically apply fixes suggested by rustc to crate `bugtest`

after fixes were automatically applied the compiler reported errors within these files:

  * src\main.rs

This likely indicates a bug in either rustc or cargo itself,
and we would appreciate a bug report! You're likely to see
a number of compiler warnings after this message which cargo
attempted to fix but failed. If you could open an issue at
https://github.com/rust-lang/rust/issues
quoting the full output of this command we'd be very appreciative!
Note that you may be able to make some more progress in the near-term
fixing code with the `--broken-code` flag

The following errors were reported:
warning: unused variable: `index`
 --> src\main.rs:3:13
  |
3 |     let mut index:usize = 0;
  |             ^^^^^ help: if this is intentional, prefix it with an underscore: `_index`
  |
  = note: `#[warn(unused_variables)]` on by default

warning: value assigned to `index` is never read
 --> src\main.rs:7:9
  |
7 |         index+=1;
  |         ^^^^^
  |
  = note: `#[warn(unused_assignments)]` on by default
  = help: maybe it is overwritten before being read?

warning: variable does not need to be mutable
 --> src\main.rs:3:9
  |
3 |     let mut index:usize = 0;
  |         ----^^^^^
  |         |
  |         help: remove this `mut`
  |
  = note: `#[warn(unused_mut)]` on by default

error[E0384]: cannot assign twice to immutable variable `index`
 --> src\main.rs:7:9
  |
5 |     for (index, v) in vector.into_iter().enumerate(){
  |          -----
  |          |
  |          first assignment to `index`
  |          help: consider making this binding mutable: `mut index`
6 |         println!("{v} {index}");
7 |         index+=1;
  |         ^^^^^^^^ cannot assign twice to immutable variable

error: aborting due to previous error; 3 warnings emitted
```

### Version

```text
rustc 1.61.0 (fe5b13d68 2022-05-18)
binary: rustc
commit-hash: fe5b13d681f25ee6474be29d748c65adcd91f69e
commit-date: 2022-05-18
host: x86_64-pc-windows-msvc
release: 1.61.0
LLVM version: 14.0.0
```

### Additional Labels

@rustbot label +I-suggestion-causes-error
